uiServices/src/testSuiteServices_test.go

```diff
@@ -191,3 +214,238 @@ func TestValidTestSuitePostConfigPathDirNotExist(t *testing.T) {
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", putTestSuites)
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/"
+	request, err := http.NewRequest(http.MethodPut, "", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, http.StatusNotFound, w.Code,  "Sucessfully updated. Should not have worked with no URL")
+}
+
+func TestPutWithNoPathSlash(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", putTestSuites)
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test"
+	request, err := http.NewRequest(http.MethodPut, "/test-suites/TestSuiteService", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t,http.StatusNoContent, w.Code,  "Did not update. Should have added '/' to path")
+}
+
+func TestNoPathTestSuitePut(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", putTestSuites)
+
+	filePath := ""
+	request, err := http.NewRequest(http.MethodPut, "/test-suites/TestSuiteService", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t,  http.StatusBadRequest, w.Code, "Successfully updated. Should not have worked due to no filepath")
+}
+
+func TestNoFileNameTestSuitePut(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	reader := strings.NewReader(validTestSuite)
+	r.HandleFunc("/test-suites", putTestSuites)
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test"
+	request, err := http.NewRequest(http.MethodPut, "/test-suites", reader)
+	request.Header.Set("testSuitePathDir", filePath)
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, http.StatusBadRequest, w.Code, "Successfully updated. Should not have worked due to no file name given")
+}
+
+
+
+
+func TestSuccessfulGetTestSuite(t *testing.T) {
+	r := chi.NewRouter()
+	r.Mount("/", GetIndexPage())
+
+	filePath := os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/"
+	request, err := http.NewRequest(http.MethodGet, "/test-suites/TestSuiteService", nil)
+
+	request.Header.Set("testSuitePathDir", filePath)
+	request.Header.Get("testSuitePathDir")
+
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, request)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.Equal(t, w.Code, http.StatusOK, "Error. Did not successfully GET")
```

Comment:
other way around. Keep your expected status to the left of your actual status code. If the test fails, the output will be misleading. Check other tests too (I've seen it in a few places)